test/TradeTrustERC721.js

```diff
@@ -1,106 +1,108 @@
-const {expect} = require("chai").use(require("chai-as-promised"));
-const ethers = require("ethers");
-
-const TitleEscrow = artifacts.require("TitleEscrow");
-const Erc721 = artifacts.require("TradeTrustERC721");
-const CalculateSelector = artifacts.require("CalculateTradeTrustERC721Selector");
+const { expect } = require("chai").use(require("chai-as-promised"));
+const chai = require("chai");
+const { solidity } = require("ethereum-waffle");
+chai.use(solidity);
 
 const assertDestroyBurntLog = (logs, tokenId) => {
   expect(logs.event).to.deep.equal("TokenBurnt");
   expect(ethers.BigNumber.from(logs.args[0].toString()).toHexString()).to.deep.equal(tokenId);
 };
 
-const assertTokenReceivedLog = (logs, operator, from, tokenId, data) => {
+const assertTokenReceivedLog = (logs, operator, from, tokenId) => {
   expect(logs.event).to.deep.equal("TokenReceived");
   expect(logs.args[0]).to.deep.equal(operator);
   expect(logs.args[1]).to.deep.equal(from);
   expect(ethers.BigNumber.from(logs.args[2].toString()).toHexString()).to.deep.equal(tokenId);
-  expect(logs.args[3]).to.deep.equal(data);
 };
 
-contract("TradeTrustErc721", accounts => {
-  const shippingLine = accounts[0];
-  const owner1 = accounts[1];
-  const owner2 = accounts[2];
-  const nonMinter = accounts[3];
-  const holder1 = accounts[4];
+describe("TradeTrustErc721", async () => {
+  let carrier1, owner1, owner2, nonMinter, holder1;
+  let TitleEscrow;
+  let Erc721;
+  let CalculateSelector;
+
+  before("", async () => {
+    let accounts = await ethers.getSigners();
+    [carrier1, owner1, owner2, nonMinter, holder1, ..._] = accounts;
+    TitleEscrow = await ethers.getContractFactory("TitleEscrow");
+    Erc721 = await ethers.getContractFactory("TradeTrustERC721");
+    CalculateSelector = await ethers.getContractFactory("CalculateTradeTrustERC721Selector");  
+  });
+
 
   const merkleRoot = "0x624d0d7ae6f44d41d368d8280856dbaac6aa29fb3b35f45b80a7c1c90032eeb3";
   const merkleRoot1 = "0x624d0d7ae6f44d41d368d8280856dbaac6aa29fb3b35f45b80a7c1c90032eeb4";
   const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
   const BURN_ADDRESS = "0x000000000000000000000000000000000000dEaD";
 
   it("should have the correct ERC165 interface support", async () => {
-    const tradeTrustERC721Instance = await Erc721.new("foo", "bar");
-    const calculatorInstance = await CalculateSelector.new();
+    const tradeTrustERC721Instance = await Erc721.connect(carrier1).deploy("foo", "bar");
+    const calculatorInstance = await CalculateSelector.connect(carrier1).deploy();
```

Comment:
this one should be dont need the (.connect()) since the contract not validating for it in the calculateSelector function. (accidentally clicked on 'Start a Review')

Reply:
oh i thought its compulsory but looks like hardhat knows how to pick an account to deploy it with even without .connect()